process/process.go

```diff
@@ -21,8 +21,8 @@ func (w *Process) Validate() error {
 		mapNode := node.GetMap()
 		if mapNode != nil {
 			for _, output := range mapNode.Outputs {
-				if ref := output.GetRef(); ref != nil {
-					if _, err := w.FindNode(ref.NodeKey); err != nil {
+				if ref := output.GetRef(); ref != nil && ref.RefKey != "" {
+					if _, err := w.FindNode(ref.RefKey); err != nil {
```

Comment:
I don't think this is right, `findNode` with check the `nodeKey`, in our case `nodeKey` is different than `refKey`.
This should be a find with a filter that the node is a task and has the same `refKey` or am I missing something here?